code/game/objects/items/defib.dm

```diff
@@ -468,17 +473,43 @@
 
 	do_help(H, user)
 
-/obj/item/shockpaddles/proc/shock_touching(dmg, mob/H)
+/// Returns TRUE when the paddles can do harmful actions, and false if they cant (If the safety is off, or if they're combat paddles)
+/obj/item/shockpaddles/proc/can_hurt()
+	if(combat)
+		return TRUE
+	if(req_defib && !defib.safety)
+		return TRUE
+	return FALSE
```

Comment:
I'm generally not a big fan of `is_thing()` or `can_do_thing()` procs that just check several conditions.
They tend to grow, be redefined for extra exceptions, and then make producing events-oriented code way harder.

My suggestion is to either make `can_hurt` a variable that changes based on the changes of the other variables (through setters), or alternatively just to turn this into a macro, which has lower chances of being expanded into:
`#define SHOCKPADDLES_CAN_HURT (combat || (req_defib && !defib.safety))`

Reply:
Good point, I'll turn it into a define